tools/md2swagger.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python
+
+# Convert lightblue metadata to swagger yaml
+#
+# This is useful in creating the first data model
+# for swagger specifications from lightblue metadata.
+# Run the script as:
+#
+#  md2swagger.py <metadata files> > model.yaml
+#
+#
+
+import sys
+import json
+import yaml
+
+def getTypeProperties(type):
+    properties=dict()
+    if type=='bigdecimal' or type=='biginteger':
+        properties['type']='string'
+    elif type=='boolean':
+        properties['type']='boolean'
+    elif type=='date':
+        properties['type']='string'
+        properties['format']='date-time'
+    elif type=='double':
+        properties['type']='number'
+        properties['format']='double'
+    elif type=='integer':
+        properties['type']='integer'
+        properties['format']='int64'
+    elif type=='string':
+        properties['type']='string'
+    return properties;
+
+def capital1(s):
+    return s[0].upper()+s[1:] 
+
+def singular(s):
+    if s[-3:]=='ies':
+        return s[0:-3]+'y'
+    elif s[-4:]=='sses':
+        return s[0:-4]+'ss'
+    elif s[-1]=='s':
+        return s[0:-1]
+    else:
+        return s;
+
+def makeEntityName(stack):
+    return ''.join(stack);
+
+def processFieldTree(fields,toplevel,nameStack):
+    properties=dict()
+    for field in fields.keys():
+        if field.find('#')==-1:
+            fieldBlock=dict();
+            fieldContents=fields[field]
+            if fieldContents['type']=='reference':
+                fieldBlock['type']='array'
+                fieldBlock['items']={'$ref':'#/definitions/'+capital1(fieldContents['entity'])}
+            elif fieldContents['type']=='object':
+                # Add a reference to the top level object
+                # For that, we have to generate a name for this class
+                fieldBlock['type']='object'
+                nameStack.append(capital1(field))
+                newEntityName=makeEntityName(nameStack)                
+                fieldBlock['items']={'$ref':'#definitions/'+newEntityName}
+                toplevel[newEntityName]=processFieldTree(fieldContents['fields'],toplevel,nameStack)
+                nameStack.pop()
+            elif fieldContents['type']=='array':
+                fieldBlock['type']='array'
+                arrayType=fieldContents['items']['type']
+                if arrayType=='object':
+                    nameStack.append(singular(capital1(field)))
+                    newEntityName=makeEntityName(nameStack)
+                    fieldBlock['items']={'$ref':'#definitions/'+newEntityName}
```

Comment:
I made some fixes in the internal repo for this you may want to grab
- need #/definitions (missing /)
- The new top level sub entities need type: object and properties: fields